fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindDynamicSplit.java

```diff
@@ -0,0 +1,196 @@
+        }
+        SlotReference slotReference = getSlotReference(logicalView.getOutput(), splitColumnName);
+        if (null == slotReference) {
+            logicalDynamicSplit.setReplaced(Boolean.FALSE);
+            LOG.warn("Can not find the split column {} in the table {}", splitColumnName, splitColumnName);
+            return result;
+        }
+        GreaterThanEqual greaterThanEqual = new GreaterThanEqual(slotReference,
+                Literal.of(range.getMinimum()));
+        LessThanEqual lessThanEqual = new LessThanEqual(slotReference, Literal.of(range.getMaximum()));
+        LogicalFilter logicalFilter = new LogicalFilter<>(ImmutableSet.of(greaterThanEqual, lessThanEqual),
+                logicalView);
+        result = result.rewriteUp(node -> {
+            if (node instanceof LogicalView) {
+                return logicalFilter;
+            }
+            return node;
+        });
+        return result;
+    }
+
+    private Plan rewritePlan(LogicalRelation logicalRelation, Plan result,
+                             LogicalDynamicSplit logicalDynamicSplit) {
+        LogicalCatalogRelation catalogRelation = convertAndCheckLogicalRelation(logicalRelation);
+        if (null == catalogRelation) {
+            return result;
+        }
+        Range range = logicalDynamicSplit.getRange();
+        SplitColumnInfo splitColumnInfo = logicalDynamicSplit.getSplitColumnInfo();
+        String splitColumnName = splitColumnInfo.getColumnName();
+        TableName splitTable = splitColumnInfo.getTableNameInfo().transferToTableName();
+        TableIf table = catalogRelation.getTable();
+        logicalDynamicSplit.setReplaced(Boolean.TRUE);
+        if (!table.getName().equals(splitTable.getTbl())) {
+            return result;
+        }
+        if (!table.getDatabase().getFullName().equals(splitTable.getDb())) {
+            return result;
+        }
+        if (!table.getDatabase().getCatalog().getName().equals(splitTable.getCtl())) {
+            return result;
+        }
+        SlotReference slotReference = getSlotReference(catalogRelation.getOutput(), splitColumnName);
+        if (null == slotReference) {
+            logicalDynamicSplit.setReplaced(Boolean.FALSE);
+            LOG.warn("Can not find the split column {} in the table {}", splitColumnName, table.getName());
+            return result;
+        }
+        GreaterThanEqual greaterThanEqual = new GreaterThanEqual(slotReference,
+                Literal.of(range.getMinimum()));
+        LessThanEqual lessThanEqual = new LessThanEqual(slotReference, Literal.of(range.getMaximum()));
+
+        LogicalFilter logicalFilter = new LogicalFilter<>(ImmutableSet.of(greaterThanEqual, lessThanEqual),
+                catalogRelation);
+        result = result.rewriteUp(node -> {
+            if (node instanceof LogicalCatalogRelation) {
+                return logicalFilter;
+            }
+            return node;
+        });
+
+        return result;
+    }
```

Comment:
can you merge this two rewritePlan functions to one? It looks mostly the same

Reply:
thx~ done